src/drivers/intel/ssp/ssp.c

```diff
@@ -98,10 +119,10 @@ static void ssp_empty_rx_fifo(struct dai *dai)
 		dai_dbg(dai, "ssp_empty_rx_fifo(), before flushing, entries %d", entries);
 
 		/* let DMA consume data or read RX FIFO directly */
-		if (direct_reads) {
-			for (i = 0; i < entries + 1; i++)
-				ssp_read(dai, SSDR);
-		}
+#if CONFIG_DMA_SUSPEND_DRAIN
+		for (i = 0; i < entries + 1; i++)
```

Comment:
can remove 106 - 108, and use:
for ( int i = 0; i < entries + 1; i++)

Reply:
Yeah, that would be nicer, but for now, we've stick to Linux kernel style and declare variable at start. I know we start to have some C++ inspired code that is not following this anymore...